### PR TITLE
Add Minitests for PS 8.0 Build Job

### PR DESCRIPTION
**Pull request by panchal-yash**
* Add Minitests for PS8.0 build job.

* Minitests have only install stage for all the supported OS after the packages are built and pushed to testing repo.

* add package_tests_ps80() function for install package tests for PS_8.0 in the build job itself along with relevant notifications to the release channel (Minitests). 

* Minitest are subset of package tests, so any failure in Minitest will not trigger package-testing and integration test jobs.

### CODE REVIEW

**Comment by EvgeniyPatlan** [comment]
I suggest reworking this in a bit different way. We can create a separate functions for running tests and installing dependencies.

I have prepared a code block but I have not tested it. Please create a test job and perform tests before merging changes


```
def installDependencies(def nodeName) {
    def aptNodes = ['min-buster-x64', 'min-bullseye-x64', 'min-bookworm-x64', 'min-bionic-x64', 'min-focal-x64', 'min-jammy-x64']
    def yumNodes = ['min-ol-8-x64', 'min-centos-7-x64', 'min-ol-9-x64', 'min-amazon-2-x64']
    
    if (aptNodes.contains(nodeName)) {
        sh '''
            sudo apt-get update
            sudo apt-get install -y ansible git wget
            ''' + (nodeName in ['min-bionic-x64', 'min-focal-x64', 'min-jammy-x64'] ? '''
            sudo apt-get install -y software-properties-common
            sudo apt-add-repository --yes --update ppa:ansible/ansible
            ''' : '')
    } else if (yumNodes.contains(nodeName)) {
        sh '''
            ''' + (nodeName == 'min-ol-8-x64' ? 'sudo yum install -y epel-release' : '') + '''
            sudo yum -y update
            sudo yum install -y ansible git wget tar
            ''' + (nodeName == 'min-amazon-2-x64' ? 'sudo amazon-linux-extras install epel' : '')
    } else {
        echo "Unexpected node name: ${nodeName}"
    }
}

def runPlaybook(def nodeName) {
    try {
        def playbook = "${ps80_install_pkg_minitests_playbook}"
        def playbook_path = "package-testing/playbooks/${playbook}"

        sh '''
            git clone --depth 1 https://github.com/Percona-QA/package-testing
        '''

        sh """
            export install_repo="\${install_repo}"
            export client_to_test="ps80"
            export check_warning="\${check_warnings}"
            export install_mysql_shell="\${install_mysql_shell}"
            ansible-playbook \
            --connection=local \
            --inventory 127.0.0.1, \
            --limit 127.0.0.1 \
            ${playbook_path}
        """
    } catch (Exception e) {
        slackNotify("${SLACKNOTIFY}", "#FF0000", "[${JOB_NAME}]: Mini Package Testing for ${nodeName} at ${BRANCH}  FAILED !!")
        error("Mini Package Tests Failed! for ${nodeName}")
    }
}

def package_tests_ps80() {
    def arrayA = [  "min-buster-x64",
                    "min-bullseye-x64",
                    "min-bookworm-x64",
                    "min-centos-7-x64",
                    "min-ol-8-x64",
                    "min-bionic-x64",
                    "min-focal-x64",
                    "min-amazon-2-x64",
                    "min-jammy-x64",
                    "min-ol-9-x64"     ]

    def stepsForParallel = [:]

    for (def nodeName : arrayA) {
        stepsForParallel[nodeName] = {
            stage("Run on ${nodeName}") {
                node(nodeName) {
                    try {
                        installDependencies(nodeName)
                    } catch (Exception e) {
                        slackNotify("${SLACKNOTIFY}", "#FF0000", "[${JOB_NAME}]: Server Provision for Mini Package Testing for ${nodeName} at ${BRANCH}  FAILED !!")
                    }
                    runPlaybook(nodeName)
                }
            }
        }
    }
    parallel stepsForParallel
}

```

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
why do you remove the existent jobs? In order to create a tests jobs you shouldn't remove prod jobs

**Review by EvgeniyPatlan — APPROVED** [review]

File: ps/jenkins/percona-server-for-mysql-8.0.groovy [hunk 1]
```diff
@@ -63,6 +63,120 @@ void cleanUpWS() {
...
+    }
+
+}
+
+def runPlaybook(def nodeName) {
+
+    def ps80_install_pkg_minitests_playbook = 'ps_80.yml'
+    def install_repo = 'testing'
+    def action_to_test = 'install'
+    def check_warnings = 'no'
```
Comment (eleo007) [inline ps/jenkins/percona-server-for-mysql-8.0.groovy:124]: Why do we skip warnings and shell checks?

File: ps/jenkins/percona-server-for-mysql-8.0.groovy [hunk 2]
```diff
@@ -656,9 +770,17 @@ parameters {
...
                 """
                 }
+
+                echo "Start Minitests for PS"
+                
+                package_tests_ps80()
+                
                 echo "Trigger Package Testing Job for PS"
+
                 build job: 'package-testing-ps80', propagate: false, wait: false, parameters: [string(name: 'product_to_test', value: 'ps80'),string(name: 'install_repo', value: "testing"),string(name: 'node_to_test', value: "all"),string(name: 'action_to_test', value: "all"),string(name: 'check_warnings', value: "yes"),string(name: 'install_mysql_shell', value: "no")]
```
Comment (eleo007) [inline ps/jenkins/percona-server-for-mysql-8.0.groovy:780]: Are we going to keep triggering the job and running the same playbook in the job?

**Review by eleo007 — APPROVED** [review]

**Review by adivinho — APPROVED** [review]

**Pull request merged**